src/software/world/team.cpp

```diff
@@ -178,6 +178,31 @@ std::vector<Robot> Team::getAllRobotsExceptGoalie() const
     return all_robots;
 }
 
+std::optional<Robot> Team::getNearestRobot(const Point& ref_point) const
+{
+    return getNearestRobot(this->getAllRobots(), ref_point);
+}
+
+std::optional<Robot> Team::getNearestRobot(const std::vector<Robot>& robots,
+                                           const Point& ref_point) const
+{
+    if (robots.empty())
+    {
+        return std::nullopt;
+    }
+
+    Robot nearest_robot = robots.at(0);
+    for (const Robot& curRobot : robots)
+    {
+        double curDistance = (ref_point - curRobot.position()).length();
+        if (curDistance < (nearest_robot.position() - ref_point).length())
+        {
+            nearest_robot = curRobot;
+        }
+    }
+
+    return nearest_robot;
```

Comment:
Nit:

Could use `std::min()` here

Reply:
Not sure how I could use `std::min()` here instead with out adding a separate function to compare two robots' distance from a point. Could you clarify how you're envisioning it?